deepcell_tracking/tracking.py

```diff
@@ -653,114 +680,24 @@ def _get_parent(self, frame, cell, predictions):
         """
         parent_id = None
         max_prob = self.division
-        for (track, cell_id), p in predictions.items():
-            prob = p[2]  # probability cell is part of the track
-            # Make sure capped tracks can't be assigned parents
-            if cell_id == cell and not self.tracks[track]['capped']:
-                # Do not call a newly-appeared sibling of "cell" a parent
-                if self.tracks[track]['frames'] == [frame]:
-                    continue
-                if prob > max_prob:
-                    parent_id, max_prob = track, prob
-        return parent_id
-
-    def _sub_area(self, X_frame, y_frame, cell_label):
-        """Fetch a neighborhood surrounding the cell in the given frame.
-
-        Slices out a neighborhood_true_size square region around the center of
-        the provided cell, and reshapes it to neighborhood_scale_size square.
 
-        Args:
-            X_frame (np.array): 2D numpy array, a frame of raw data.
-            y_frame (np.array): 2D numpy array, a frame of annotated data.
-            cell_label (int): The label of the cell to slice out.
-
-        Returns:
-            numpy.array: the resized region of X_frame around cell_label.
-        """
-        pads = ((self.neighborhood_true_size, self.neighborhood_true_size),
-                (self.neighborhood_true_size, self.neighborhood_true_size),
-                (0, 0))
+        predictions = predictions_dict['predictions']
 
-        X_padded = np.pad(X_frame, pads, mode='constant', constant_values=0)
-        y_padded = np.pad(y_frame, pads, mode='constant', constant_values=0)
+        for track_id in range(predictions.shape[0]):
+            for cell_idx in range(predictions.shape[1]):
+                # cell_id = self.idx_to_id[(frame, cell_idx)]
+                # prob cell is part of the track
 
-        roi = (y_padded == cell_label).astype('int32')
-        props = regionprops(np.squeeze(roi), coordinates='rc')
+                prob = predictions_dict['predictions'][track_id, cell_idx, 2]
```

Comment:
We already have the array as a variable, let's use it.

```suggestion
                prob = predictions[track_id, cell_idx, 2]
```